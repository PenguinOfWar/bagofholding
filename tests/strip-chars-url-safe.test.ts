import stripCharsUrlSafe from '@/packages/strip-chars-url-safe/strip-chars-url-safe';

it('converts a string into a url safe slug', () => {
  const string = stripCharsUrlSafe('Welcome: to my blog!12!!');

  expect(string).toMatchSnapshot();
});

it('handles nasty strings gracefully', () => {
  const string = stripCharsUrlSafe(
    'Welcome: to my Iñtërnâtiônàlizætiøn blog about Internationalization'
  );

  expect(string).toBe(
    'welcome-to-my-itrntinliztin-blog-about-internationalization'
  );
  expect(string).toMatchSnapshot();
});
