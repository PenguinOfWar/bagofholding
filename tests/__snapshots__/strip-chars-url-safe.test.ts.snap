// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts a string into a url safe slug 1`] = `"welcome-to-my-blog12"`;

exports[`handles nasty strings gracefully 1`] = `"welcome-to-my-itrntinliztin-blog-about-internationalization"`;
